app/src/main/java/ch/epfl/sdp/drone3d/mission/Parallelogram.kt

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (C) 2021  Drone3D-Team
+ * The license can be found in LICENSE at root of the repository
+ */
+
+package ch.epfl.sdp.drone3d.mission
+
+import ch.epfl.sdp.drone3d.mission.Point.Companion.distance
+import com.mapbox.mapboxsdk.geometry.LatLng
+
+/**
+ * Utility class to represent a parallelogram
+ */
+data class Parallelogram(val origin:Point, val dir1Span:Vector, val dir2Span:Vector){
+
+    constructor(origin: Point,dir1Point:Point,dir2Point:Point): this(origin,Vector(origin,dir1Point),Vector(origin,dir2Point))
```

Comment:
Maybe rename dir1Point and dir2Point to adjacentPoint1 and adjacentPoint2

Reply:
Good point, done